soroban-sdk/src/storage.rs

```diff
@@ -76,18 +76,59 @@ impl Storage {
         Storage { env: env.clone() }
     }
 
+    /// Storage for data that can stay in the ledger forever until deleted.
+    ///
+    /// Persistent entries might be 'evicted' from the ledger if they run out
+    /// of the rent balance. However, evicted entries can be restored and also
+    /// they cannot be recreated. This means that semantically these entries
+    /// behave 'as if' they were stored in the ledger forever.
+    ///
+    /// This should be used for data that requires persistency, such as token
+    /// balances, user properties etc.
     pub fn persistent(&self) -> Persistent {
         Persistent {
             storage: self.clone(),
         }
     }
 
+    /// Storage for data that may stay in ledger only for a limited amount of
+    /// time.
+    ///
+    /// This is cheaper than `persistent()` storage.
+    ///
+    /// Temporary entries will be removed from the ledger after their lifetime
+    /// ends. Removed entries can be created again, potentially with different
+    /// values.
+    ///
+    /// This should be used for data that needs to only exist for a limited
+    /// period of time, such as oracle data, claimable balances, offerc etc.
     pub fn temporary(&self) -> Temporary {
         Temporary {
             storage: self.clone(),
         }
     }
 
+    /// Storage for a **small amount** of persistent data associated with
+    /// the current contract's instance.
+    ///
+    /// All the `instance` entries are stored in a single `persistent()` ledger
+    /// entry. That makes it cheap to use this, but also limits the total
+    /// size of the stored data.
```

Comment:
We're discussing implementation details. We should discuss attributes / properties of instance storage instead: loaded automatically and doesn't appear in footprints, increases the cost of every invocation of the contract whether the data is used or not.

We discuss limits here, but don't define any concrete limits. If we discuss the attributes / properties ☝🏻 it might be clearer if we follow with, "therefore due to this qualities of instance data, it is really only suitable for small data."

Reply:
I've tried to rephrase this a bit. There is a lot of information to convey; I'm still not sure how to structure this in the most efficient fashion.

Reply:
I think this is really good. There's a lot to communicate, I think the comment does a good job of picking and choosing.